File: RADWIMPS.go
package main

import "fmt"

type RADWIMPS struct {}

func (r RADWIMPS) then() RADWIMPS {
    fmt.Print("前")
    return r;
}

func (r RADWIMPS) 世() RADWIMPS {
    fmt.Println("世")
    return r;
}

func main() {
    RADWIMPS := RADWIMPS {}
    RADWIMPS.then().then().then().世()
}

File: RADWIMPS.m
#include <Block.h>
#include <Foundation/NSObject.h>
#include <stdio.h>

@interface RADWIMPS : NSObject

@property(readonly, nonatomic, copy) RADWIMPS * (^then)();
@property(readonly, nonatomic, copy) void (^世)();

@end

@implementation RADWIMPS

- (RADWIMPS * (^)())then {
  return Block_copy(^RADWIMPS *() {
    printf("前");
    return self;
  });
}

- (void (^)())世 {
  return ^() {
    printf("世\n");
  };
}

@end

int main() {
  RADWIMPS *radwimps = [[RADWIMPS alloc] init];

  radwimps.then().then().then().世();

  [radwimps dealloc];
}
